src/prommis/solvent_extraction/sx_model_flowsheet_dynamic.py

```diff
@@ -0,0 +1,186 @@
+""" 
+Demonstration flowsheet for dynamic state solvent extraction loading process
+using parameters and data derived from West Kentucky No. 13 coal refuse.
+
+Authors: Arkoprabho Dasgupta
+
+"""
+
+from pyomo.environ import ConcreteModel, SolverFactory, units, TransformationFactory
+
+from idaes.core import FlowDirection, FlowsheetBlock
+from idaes.core.initialization import InitializationStatus
+from idaes.core.initialization.block_triangularization import (
+    BlockTriangularizationInitializer,
+)
+
+from prommis.leaching.leach_solution_properties import LeachSolutionParameters
+from prommis.solvent_extraction.ree_og_distribution import REESolExOgParameters
+from prommis.solvent_extraction.solvent_extraction import SolventExtraction
+
+"""
+Method of building a dynamic solvent extraction model with a specified number of 
+stages and with two separate property packages for the two inlet streams.
+This is a loading operation, so no additional argument has to be specified.
+
+"""
+
+m = ConcreteModel()
+
+time_duration = 60
+
+m.fs = FlowsheetBlock(dynamic=True, time_set=[0, time_duration], time_units=units.hour)
+
+m.fs.prop_o = REESolExOgParameters()
+m.fs.leach_soln = LeachSolutionParameters()
+
+number_of_stages = 3
+
+m.fs.solex = SolventExtraction(
+    number_of_finite_elements=number_of_stages,
+    aqueous_stream={
+        "property_package": m.fs.leach_soln,
+        "flow_direction": FlowDirection.forward,
+        "has_energy_balance": False,
+        "has_pressure_balance": False,
+    },
+    organic_stream={
+        "property_package": m.fs.prop_o,
+        "flow_direction": FlowDirection.backward,
+        "has_energy_balance": False,
+        "has_pressure_balance": False,
+    },
+)
+
+"""
+Discretization of the time domain, and specification of the partition coefficients,
+volume, volume fractions, and the initial conditions of state variables for the components
+for all the stages. 
+
+"""
+
+m.discretizer = TransformationFactory("dae.finite_difference")
+m.discretizer.apply_to(m, nfe=5, wrt=m.fs.time, scheme="BACKWARD")
```

Comment:
1. Is a backward finite difference method the correct one to chose for the time domain? I think you normally use a forward difference (implicit Euler) method for time domains.
2. Is 5 time points sufficient? That seems very few points to be using in an example.

Reply:
1. I will use Forward
2. I just gave 5 to check because it was going very long for more than 5 units.

Reply:
Dynamic models always take a while to run. The number of time points needs to be matched to the time constant of the process.

Reply:
As discussed with you, I have given OCFE, and it is working, and the number of time points is more now.